pkg/cluster/check/events.go

```diff
@@ -0,0 +1,93 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package check
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sort"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	machineapi "github.com/talos-systems/talos/pkg/machinery/api/machine"
+	"github.com/talos-systems/talos/pkg/machinery/client"
+)
+
+// NodeBootedAssertion checks whether nodes reached end of 'Boot' sequence.
+//nolint: gocyclo
+func NodeBootedAssertion(ctx context.Context, cluster ClusterInfo) error {
```

Comment:
Could we call this AllNodesBooted or ClusterNodesBootod to better reflect its multi-nodal operation?

Reply:
renamed to `AllNodesBootedAssertion`